lib/plugins/aws/package/compile/events/cloudFront/index.js

```diff
@@ -0,0 +1,284 @@
+      });
+      return originObj;
+    }
+
+    this.serverless.service.getAllFunctions().forEach((functionName) => {
+      const functionObj = this.serverless.service.getFunction(functionName);
+      if (functionObj.events) {
+        functionObj.events.forEach(event => {
+          if (event.cloudFront) {
+            if (_.isObject(event.cloudFront) === false || _.isArray(event.cloudFront)) {
+              throw new Error('cloudFront event has to be an object');
+            }
+            const lambdaVersionLogicalId = _.findKey(Resources, {
+              Type: 'AWS::Lambda::Version',
+              Properties: {
+                FunctionName: {
+                  Ref: this.provider.naming.getLambdaLogicalId(functionName),
+                },
+              },
+            });
+
+            const pathPattern =
+              typeof event.cloudFront.pathPattern === 'string'
+                ? event.cloudFront.pathPattern
+                : undefined;
+
+            let origin =
+              createOrigin(functionName, event.cloudFront.origin, this.provider.naming);
+
+            const existingOrigin =
+              _.find(origins, o => o.OriginPath === origin.OriginPath);
+
+            if (typeof existingOrigin === 'undefined') {
+              origins.push(origin);
+            } else {
+              origin = extendDeep(existingOrigin, origin);
+            }
+
+            let behavior = {
+              ViewerProtocolPolicy: 'allow-all',
+              TargetOriginId: origin.Id,
+              ForwardedValues: {
+                QueryString: false,
+              },
+              LambdaFunctionAssociations: [{
+                EventType: event.cloudFront.eventType,
+                LambdaFunctionARN: {
+                  Ref: lambdaVersionLogicalId,
+                },
+              }],
+            };
+
+            if (typeof pathPattern !== 'undefined') {
+              _.merge(behavior, { PathPattern: pathPattern });
+            }
+
+            const existingBehaviour =
+              _.find(behaviors, o =>
+                o.PathPattern === behavior.PathPattern
+                  && o.TargetOriginId === behavior.TargetOriginId);
+
+            if (typeof existingBehaviour === 'undefined') {
+              behaviors.push(behavior);
+            } else {
+              behavior = extendDeep(existingBehaviour, behavior);
+            }
+
+            lambdaAtEdgeFunctions.push(_.merge({},
+              functionObj, { functionName, lambdaVersionLogicalId }));
+          }
+        });
+      }
+    });
+
+    // sort that first is without PathPattern if available
+    behaviors.sort((a, b) => {
+      if (b.PathPattern) {
+        return -1;
+      }
+      return 0;
+    });
+
+    if (lambdaAtEdgeFunctions.length > 0) {
+      if (this.provider.getRegion() !== 'us-east-1') {
+        throw new
+          Error('CloudFront associated functions have to be deployed to the us-east-1 region.');
```

Comment:
Currently, this functionality can't work other than us-east-1 due to restriction on AWS end.
But I think this check is unnecessary since we need to update this check if AWS supports the other regions in the future.

Reply:
Yes, all the CloudFront related features, lambda@edge, certificates, etc. are currently defined in and replicated from us-east-1. I'm not sure if AWS is going to change that anytime soon.